shuffle/move/examples/main/helpers.ts

```diff
@@ -61,37 +63,104 @@ export function buildScriptFunctionTransaction(
-    args,
+  privateKeyBytes = normalizePrivateKey(privateKeyBytes);
+  const [sfAddress, sfModule, sfFunction] = splitFullyQualifiedName(
+    scriptFunction,
   );
+  const typeStructs = splitTypes(types);
 
-  // TODO(dimroc) : Help clean this up
-  const shuffleDir = Deno.env.get("SHUFFLE_HOME") || "unknown";
-  const privateKeyPath = path.join(shuffleDir, "accounts/latest/dev.key");
-  const senderAddressPath = path.join(shuffleDir, "accounts/latest/address");
-  const senderAddress = await Deno.readTextFile(senderAddressPath);
-  const fullSenderAddress = "0x" + senderAddress;
-  const privateKeyBytes = await Deno.readFile(privateKeyPath);
-  return await buildAndSubmitTransaction(
-    fullSenderAddress,
-    sequenceNumber,
-    privateKeyBytes,
-    payload,
-  );
+  // deno-lint-ignore no-explicit-any
+  const request: any = {
+    "sender": addressStr,
+    "sequence_number": `${sequenceNumber}`,
+    "max_gas_amount": "1000000",
+    "gas_unit_price": "0",
+    "gas_currency_code": "XUS",
+    "expiration_timestamp_secs": "99999999999",
+    "payload": {
+      "type": "script_function_payload",
+      "module": {
+        "address": sfAddress,
+        "name": sfModule,
+      },
+      "function": sfFunction,
+      "type_arguments": typeStructs,
+      "arguments": normalizeScriptFunctionArgs(args),
+    },
+  };
+
+  const resp = await fetch(relativeUrl("/transactions/signing_message"), {
+    method: "POST",
+    body: JSON.stringify(request),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+
+  if (!isSuccess(resp.status)) {
+    const errorMsg = (await resp.json()).message;
+    throw `unable to create signing message: ${resp.statusText}. ${errorMsg}`;
+  }
+
+  const signingMsgPayload = await resp.json();
+  const signingMsg = signingMsgPayload.message.slice(2); // remove 0x prefix
+
+  const publicKey = bufferToHex(await ed.getPublicKey(privateKeyBytes));
+  const signature = await ed.sign(signingMsg, privateKeyBytes);
+  request.signature = {
+    "type": "ed25519_signature",
+    "public_key": publicKey,
+    "signature": signature,
+  };
+
+  // deno-lint-ignore no-explicit-any
+  const txnResponse: any = await fetch(relativeUrl("/transactions"), {
+    method: "POST",
+    body: JSON.stringify(request),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  if (!isSuccess(txnResponse.status)) {
+    const errorMsg = (await txnResponse.json()).message;
+    throw `unable to create signing message: ${txnResponse.statusText}. ${errorMsg}`;
+  }
+  return await txnResponse.json();
+}
+
+function isSuccess(status: number) {
+  return status >= 200 && status < 300;
+}
+
+function splitFullyQualifiedName(
+  scriptFunction: string,
+): [string, string, string] {
+  const [addr, moduleName, functionName]: string[] = scriptFunction.split("::");
+  return [addr, moduleName, functionName];
+}
+
+// deno-lint-ignore ban-types
+function splitTypes(types: string[]): object[] {
```

Comment:
We accepts full name "0x1::XDX::XDX" now, this convert probably will end with error

Reply:
Ah, because of the module type object below used to determine the script function, I thought that types supported both approaches. I'm all for just sending up the string and having the server do the hard work.

```json
    "payload": {
      "type": "script_function_payload",
      "module": {
        "address": sfAddress,
        "name": sfModule,
```